share/spack/docker/ci/Dockerfile

```diff
@@ -0,0 +1,74 @@
+ARG BASE
+
+FROM $BASE
+MAINTAINER Spack Maintainers <maintainers@spack.io>
+
+ENV CURRENTLY_BUILDING_DOCKER_IMAGE=1
+
+MASK PUSH
+MASK [[ $DISTRO == centos ]]
+RUN yum update -y                                          \
+ && yum install -y gmp-devel libmpc-devel mpfr-devel which \
```

Comment:
I have found it useful to have an editor of some kind in these images, so I had installed `vim`.  Do you have some suggestion for a replacement?  Maybe something is already there and I just didn't know about it?

Reply:
I prefer to avoid preinstalling any EDITORS in the main images, but I can get behind doing this for the CI builder images.